### ISSUE TITLE
error creating router_v2 using external_fixed_ip and set_router_gateway_after_create

### ISSUE THREAD

**Issue by chrischdi** [terraform-providers/terraform-provider-openstack#317]
Hi there,

Thank you for opening an issue. Please note that we try to keep the Terraform issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

### Terraform Version
```
$ terraform -v
Terraform v0.11.7
+ provider.openstack v1.4.0
```

### Affected Resource(s)
- resource_openstack_networking_router_v2

### Terraform Configuration Files
```hcl
provider "openstack" {
}
data "openstack_networking_network_v2" "extnet" {
  name = "ext_net_os1pi010"
}

data "openstack_networking_subnet_v2" "extsubnet" {
  name = "ext_net_os1pi010"
}

resource "openstack_networking_router_v2" "router01" {
  name = "router-fixed-tf01"

  external_network_id = "${data.openstack_networking_network_v2.extnet.id}"

  external_fixed_ip {
    subnet_id = "${data.openstack_networking_subnet_v2.extsubnet.id}"
    ip_address = "53.48.0.171"
  }
  value_specs {
    router_type = "exclusive"
  }
  vendor_options {
    set_router_gateway_after_create = true
  }
}
```

### Debug Output
See https://gist.github.com/chrischdi/9851e05088988ef42076b0cb61de4e30

### Panic Output
See https://gist.github.com/chrischdi/7ed324622ee89a2ec9f6eaa51b1e67fe

### Expected Behavior
A router should have been created having the ip_address `53.48.0.171` assigned to its port inside the external network

### Actual Behavior
Terraform panics

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform init`
1. `terraform apply -auto-approve`

### Important Factoids
`openstack/resource_openstack_networking_router_v2.go` contains a bug. 
When `set_router_gateway_after_create` is `true` and `enable_snat` and/or `external_network_id` is set the code will panic because it tries to set variables at `createOpts.GatewayInfo` which is not initialized (see [enable_snat](https://github.com/terraform-providers/terraform-provider-openstack/blob/master/openstack/resource_openstack_networking_router_v2.go#L187) and [external_fixed_ip](https://github.com/terraform-providers/terraform-provider-openstack/blob/master/openstack/resource_openstack_networking_router_v2.go#L195)).

The code does not handle the `vendorUpdateGateway` variable for `enable_snat` and `external_fixed_ip`

Also a update does not handle `enable_snat` or `external_fixed_ip` when `set_router_gateway_after_create` is set.

### References
We have a working patch and addded some test cases:
https://gist.github.com/chrischdi/3025610527f0d4da870c9fa006d267cb

We can also provide a PR for this bug.

A test for the `external_fixed_ip` setup would require to know a free external ip.
I think this could be done by fetching a IP address via gophercloud but this would obviously require more work to be done.

**Comment by fatmcgav** [comment]
@chrischdi Cheers for taking the time to report this issue, and also come up with a patch to fix it... 

We'd definitely welcome a PR with the appropriate changes. From a quick review, the patch looks sensible.

**Comment by chrischdi** [comment]
Fixes by Mercedes PR #309
